src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -55,9 +59,16 @@ public void setPayloadDecoder(Decoder<?> payloadDecoder) {
 		this.payloadDecoder = payloadDecoder;
 	}
 
+	public void setAutoCommitOffset(boolean autoCommitOffset) {
+		this.autoCommitOffset = autoCommitOffset;
+	}
+
 	@Override
 	protected void onInit() {
-		this.messageListenerContainer.setMessageListener(new ChannelForwardingMessageListener());
+		this.messageListenerContainer.setAutoCommitOffset(autoCommitOffset);
```

Comment:
(And I agree about this being wrong, as there's no telling in which order the two set methods are called. So I'm onboard on getting rid of the autoCommitOffset setting. In fact, we can just push that check down to the `QueueingMessageListenerInvoker`.